crates/sui/src/client_commands.rs

```diff
@@ -615,6 +665,54 @@ impl SuiClientCommands {
         context: &mut WalletContext,
     ) -> Result<SuiClientCommandResult, anyhow::Error> {
         let ret = Ok(match self {
+            SuiClientCommands::ReplayTransaction { rpc_url, tx_digest } => {
+                let cmd = ReplayToolCommand::ReplayTransaction {
+                    tx_digest,
+                    show_effects: true,
+                    diag: false,
+                    executor_version_override: None,
+                    protocol_version_override: None,
+                };
+                // todo: automatically get non pruning fullnode url
+                let _command_result =
+                    sui_replay::execute_replay_command(Some(rpc_url), false, false, None, cmd)
+                        .await;
+                SuiClientCommandResult::ReplayTransaction
+            }
+            SuiClientCommands::ReplayBatch {
+                rpc_url,
+                path,
+                terminate_early,
+            } => {
+                let cmd = ReplayToolCommand::ReplayBatch {
+                    path,
+                    terminate_early,
+                    batch_size: 16,
+                };
+                // todo: automatically get non pruning fullnode url
+                let _command_result =
+                    sui_replay::execute_replay_command(Some(rpc_url), false, false, None, cmd)
+                        .await;
+                SuiClientCommandResult::ReplayBatch
+            }
+            SuiClientCommands::ReplayCheckpoints {
+                rpc_url,
+                start,
+                end,
+                terminate_early,
+            } => {
+                let cmd = ReplayToolCommand::ReplayCheckpoints {
+                    start,
+                    end,
+                    terminate_early,
+                    max_tasks: 16,
+                };
+                // todo: automatically get non pruning fullnode url
+                let _command_result =
```

Comment:
You don't need a non-pruning FN to replay. You just need a FN which contains the data as at the time of the TX executed.
For most users, replaying within the past few epochs is sufficient, so many FNs will do.